lib/services/elasticsearch.js

```diff
@@ -1585,200 +1557,133 @@ class ElasticSearch extends Service {
+    const defer = function defer () {
+      let
+        resolve,
+        reject;
+
+      const promise = new Bluebird((res, rej) => {
+        resolve = res;
+        reject = rej;
+      });
 
-  if (index) {
-    data.index = index;
-  }
+      return { resolve, reject, promise };
+    };
 
-  if (request.input.resource.collection) {
-    data.type = request.input.resource.collection;
+    return new ESClient({ defer, ...this.config.client });
   }
 
-  for (const name of extraParams) {
-    if (typeof request.input.args[name] !== 'undefined') {
-      data[name] = request.input.args[name];
+  /**
+   * Gets the name of an esIndex for an index + collection
+   *
+   * @param {String} index
+   * @param {String} collection
+   *
+   * @returns {String} esIndex name (eg: '&nepali#liia')
+   */
+  _getESIndex (index, collection) {
+    if (! index || ! collection) {
+      return null;
     }
-  }
-
-  return data;
-}
 
-/**
- * Extracts the index from the provided request.
- * Throws if it refers to an internal index
- *
- * @param {Request} request
- * @return {string}
- * @throws {BadRequestError}
- */
-function getIndex(request) {
-  if (request.input.resource.index && request.input.resource.index[0] === '%') {
-    errorsManager.throw(
-      'external',
-      'elasticsearch',
-      'index_reserved',
-      request.input.resource.index
-    );
+    return `${this.indexPrefix}${index}${NAME_SEPARATOR}${collection}`;
   }
 
-  return request.input.resource.index;
-}
-
-/**
- * Scroll index in elasticsearch and return all document ids that match the filter
- *
- * @param {object} client - elasticsearch client
- * @param {object} esRequest
- * @returns {Promise} resolve to an array of documents IDs
- */
-function getAllIdsFromQuery(client, esRequest) {
-  const ids = [];
+  /**
+   * Extracts the index name from esIndex name
+   *
+   * @param {String} esIndex
+   *
+   * @returns {String} index name
+   */
+  _extractIndex (esIndex) {
+    return esIndex.substr(1, esIndex.indexOf(NAME_SEPARATOR) - 1);
+  }
 
-  return new Bluebird((resolve, reject) => {
-    client.search(esRequest, function getMoreUntilDone(error, response) {
-      if (error) {
-        return reject(error);
-      }
+  /**
+   * Extracts the collection name from esIndex name
+   *
+   * @param {String} esIndex
+   *
+   * @returns {String} collection name
+   */
+  _extractCollection (esIndex) {
+    const separatorPos = esIndex.indexOf(NAME_SEPARATOR);
 
-      response.hits.hits.forEach(hit => ids.push(hit._id));
+    return esIndex.substr(separatorPos + 1, separatorPos - 1);
+  }
 
-      if (response.hits.total !== ids.length) {
-        client.scroll({
-          scrollId: response._scroll_id,
-          scroll: esRequest.scroll
-        }, getMoreUntilDone);
-      }
-      else {
-        resolve(ids);
-      }
-    });
-  });
-}
+  /**
+   * Returns a list of index names from esIndex names
+   * By default, return the names of 'user' indexes.
```

Comment:
```suggestion
   * By default, return the names of 'public' indexes.
```

But reading the code, this comment seems off.